CONTRIBUTING.md

```diff
@@ -179,3 +179,60 @@ Do your best to follow the clean code guidelines.
 - Don’t return an error code, throw an exception instead.
 
 Ref: Book: Clean Code by Robert C. Martin (Uncle Bob)
+
+## Compilation and Installation
+
+This guide is helpful to compile and install Nmstate from source, for installing stable release or other installation methods, pleaser refer to [Nmstate installation guide](https://nmstate.io/user/install.md)
+
+### Clone the source
+```bash
+git clone https://github.com/nmstate/nmstate.git
+cd nmstate
+```
+
+#### Install build tool
+```bash
+# Install Cargo using rustup (recommended for development)
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source $HOME/.cargo/env
+cargo --version
+
+# Alternatively, install Cargo from your distribution's repositories
+sudo dnf install cargo  # Fedora, RHEL (using dnf)
+sudo yum install cargo  # Legacy RHEL (using yum)
+sudo apt-get install cargo  # Ubuntu, Debian
+```
+Note:
+  - Rust developers are encouraged to install Cargo using rustup, which is recommended for development purposes.
+  - Instructions for installing Cargo from distribution repositories are provided as an alternative for those who prefer that method.
+  - Distribution-specific commands for installing Cargo are included for Fedora (using dnf), Legacy RHEL (using yum), Ubuntu, and Debian.
```

Comment:
This is a very long explanation for something that it's very simple. Also, I insist in that we don't need to provide the exact command to install from every distribution. I would just put something like:

> Install cargo using rustup (recommended): URL to official website
> Alternatively, you can install cargo from your distribution's repositories.